src/Log/LogService.php

```diff
@@ -417,11 +422,19 @@ public function getDefaultFormat()
             $this->defaultFormat = $format;
         }
 
-        $runId = $this->getRunId();
-
-        $format = rtrim(preg_replace('/%runId%/', $runId, $this->defaultFormat), PHP_EOL) . PHP_EOL;
+        return $this->defaultFormat;
+    }
 
-        return $format;
+    /**
+     * Add RunId to format string.
+     *
+     * @param $format string
+     * @returns string
+     */
+    public function replaceFormatPlaceholder($format)
```

Comment:
Better, but not great. Would have accepted it, if there hadn't been any other issues. If you just saw the name of the function without knowing anything else, what would you think it does? Naming stuff is important. The function is "preparing the format". It is "processing the placeholders". It is also taking care of `PHP_EOL`. I am just thinking out loud. Maybe {{prepareFormat}} would be enough and easier to understand?

Reply:
`prepareFormat()` sounds good.